src/block_library_controller.js

```diff
@@ -319,3 +324,61 @@ BlockLibraryController.prototype.updateButtons = function(savedChanges) {
   var isInLibrary = this.has(blockType);
   this.view.updateButtons(blockType, isInLibrary, savedChanges);
 };
+
+/**
+ * Return a string of all blockTypes for building a library JSON
```

Comment:
Originally the entirety of the library was made in to a string in makeLibraryJSON then externally parsed in to JSON (which made little sense//was the subject of the comment). I took the loop that creates the text for blockTypes and put it in to a separate function. This was because I'm currently making the tree such that it updates along with in-session additions and deletions within the library, and in order to do so need to add to the string that precedes the blockTypes/children (now called 'head' in makeLibraryJSON) as well as what comes after it (now called 'tail', also in makeLibraryJSON) and it got a little obnoxious to read. Additionally, with implementing multiple projects I'm anticipating a structure wherein project titles are head nodes and the blockTypes contained within them are their children. Thus this function will make it easier to create the requisite data segments for the nodes within a project, allowing me to keep the process of making this file as readable/easy as possible (I'm thinking just having a for loop running through each project and concatenating that project's title and the result of this function). It's still a string in this phase because it doesn't comprise a complete JSON file for creating a tree, and gets put together in makeLibraryJSON (which returns the completed object). I'm also anticipating that this function, by virtue of not being a complete object/ only populating children, will make in-session adding of projects with blocks in them already (importing a project is the only user behavior I think would require this) easier/more logical.